datafusion/core/src/datasource/mod.rs

```diff
@@ -155,11 +155,14 @@ pub async fn get_statistics_with_limit(
 
 fn create_max_min_accs(
     schema: &Schema,
-) -> (Vec<Option<MaxAccumulator>>, Vec<Option<MinAccumulator>>) {
-    let max_values: Vec<Option<MaxAccumulator>> = schema
+) -> (
```

Comment:
Likewise the change in this file does not make sense to me -- why is a sliding accumulator needed (only) for calculating minimums?